File: ABC025/B.rb
n, a, b = gets.chomp.split(' ').map(&:to_i)
cur = 0
n.times do
  s, ds = gets.chomp.split(' ')
  d = ds.to_i
  if d < a
    d = a
  elsif b < d
    d = b
  end
  d *= -1 if s == 'East'
  cur += d
end

if cur.zero?
  puts 0
elsif cur.negative?
  puts "East #{-cur}"
else
  puts "West #{cur}"
end

File: ABC025/B.cpp
#include <bits/stdc++.h>

using namespace std;

int main (void) {
    int N, A, B;
    cin >> N >> A >> B;

    int cur = 0;
    for ( int i = 0; i < N; i++ ) {
        string s;
        int d;
        cin >> s >> d;

        d = max(A, min(B, d));
        if ( s == "East" ) d *= -1;
        cur += d;
    }

    if ( cur == 0 ) cout << 0 << endl;
    else cout << ((cur < 0) ? "East " : "West ") << abs(cur) << endl;
    return 0;
}
